rasa/graph_components/adders/nlu_prediction_to_history_adder.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+from rasa.shared.core.events import UserUttered
+from typing import Dict, Text, Any, Optional, List
+
+from rasa.core.channels.channel import UserMessage
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.nlu.constants import ENTITIES, INTENT, TEXT
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.core.trackers import DialogueStateTracker
+from rasa.shared.core.domain import Domain
+
+
+class NLUPredictionToHistoryAdder(GraphComponent):
+    """Adds NLU predictions to DialogueStateTracker."""
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> NLUPredictionToHistoryAdder:
+        """Creates component (see parent class for full docstring)."""
+        return cls()
+
+    # Should this be predict since it's run during inference?
+    def add(
+        predictions: List["Message"],
+        dialogue_tracker: "DialogueStateTracker",
+        domain: Domain,
+        original_message: UserMessage,
+    ) -> "DialogueStateTracker":
+        for message in predictions:
+            # Need to find out where to get time from
+            # Message contains time but it is an int, we want a float
+            # What is domain for?
+            user_utterance = UserUttered(
+                message.data.get(TEXT),
+                message.data.get(INTENT),
+                message.data.get(ENTITIES),
+                message.data,
+                None,
```

Comment:
Is this fine to leave as None - it's the timestamp. `Message` gives us an `int` timestamp but this expects `float`, is there a conversion?

Reply:
@usc-m I'd exactly do what we currently do [here](https://github.com/RasaHQ/rasa/blob/f58acdea8408563c428282d3159b11d3973acacd/rasa/core/processor.py#L598). This means
- use `tracker.update(event, domain)`
- don't pass a timestamp 

Might also be worth to copy over the logging statements from the `processor` code.

Reply:
+1 for removing timestamp and using keyword arguments for `input_channel`, `message_id`, `metadata`